project/source/Scripts/Objects/LabContainer.gd

```diff
@@ -4,43 +4,43 @@ class_name LabContainer
 
 # This container models a container (like a flask) that can be emptied of its contents.
 
-var contents = []
+var contents: Array[Substance] = []
 
-func LabObjectReady():
+func LabObjectReady() -> void:
 	update_display()
 
-func TryInteract(others):
-	for other in others:
-		if(other.is_in_group('Container')):
+func TryInteract(other_substance: Array[Substance]) -> bool:
```

Comment:
I think `other` is probably `Array[LabObject]`. As far as I can tell, `Substance`s are only ever passed around and never directly interacted with.

Reply:
That makes sense. I changed the type of other to be of Array[LabObject]